test/functional/test_bmc_peer_communication.py

```diff
@@ -0,0 +1,240 @@
+    global ivn_file
+    os.environ['PATH'] = new_path
+    ivn_file = saved_config_file()
+
+
+def teardown_module():
+    global ivn_file
+    topo = Topology(ivn_file)
+    topo.delete()
+    os.unlink(ivn_file)
+    os.environ['PATH'] = old_path
+
+
+class test_bmc_communication(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        pass
+
+    @classmethod
+    def tearDownClass(cls):
+        global fake_node0
+        global fake_node1
+        if fake_node0:
+            test_bmc_communication._stop_node(fake_node0)
+        if fake_node1:
+            test_bmc_communication._stop_node(fake_node1)
+
+    @staticmethod
+    def _stop_node(node_obj):
+        node_obj.stop()
+        node_obj.terminate_workspace()
+
+    def _start_node(self, node_info):
+        fake_node_obj = CNode(node_info)
+        fake_node_obj.init()
+        fake_node_obj.precheck()
+        fake_node_obj.start()
+        return fake_node_obj
+
+    def node_cfg_up(self, node_name, ns_name, boot_image, node_type):
+        global add_content0
+        global add_content1
+        global emu_file_for_del0
+        global emu_file_for_del1
+        fake_node = None
+        fake_node = copy.deepcopy(FakeConfig().get_node_info())
+
+        emu_dir = os.path.join(config.infrasim_data, node_type)
+        emu_file = os.path.join(emu_dir, node_type + ".emu")
+        if ns_name == "node0ns":
+            for i in add_content0:
+                os.system("echo {} >> {}".format(i, emu_file))
+        if ns_name == "node1ns":
+            for i in add_content1:
+                os.system("echo {} >> {}".format(i, emu_file))
+
+        fake_node['name'] = node_name
+        fake_node['type'] = node_type
+        fake_node['namespace'] = ns_name
+        fake_node['compute']['storage_backend'][0]["drives"][0]["file"] = boot_image
+        fake_node["compute"]["networks"][0]["port_forward"] = [{"outside": 8022, "inside": 22, "protocal": "tcp"}]
+        fake_node["bmc"] = {}
+        fake_node["bmc"]["peer-bmcs"] = [
+            {
+                "interface": "lanplus",
+                "user": "admin",
+                "password": "admin"
+            }
+        ]
+
+        if node_name == "test0":
+            fake_node["bmc"]["peer-bmcs"][0]["addr"] = 0x1e
+            fake_node["bmc"]["peer-bmcs"][0]["host"] = "192.168.188.92"
+        else:
+            fake_node["bmc"]["peer-bmcs"][0]["addr"] = 0x1c
+            fake_node["bmc"]["peer-bmcs"][0]["host"] = "192.168.188.91"
+
+        fake_node_up = self._start_node(fake_node)
+        return fake_node_up, emu_file
+
+    def client_ssh(self, ns_ip):
+        ssh = sshclient.SSH(host=ns_ip, username="root", password="root", port=8022)
+        ssh.wait_for_host_up()
+        return ssh
+
+    def test_peer_bmc_communication(self):
+        global cmd_prefix0
+        global cmd_prefix1
+        global ivn_file
+        global fake_node0
+        global fake_node1
+        topo = Topology(ivn_file)
```

Comment:
move Topology.create to setup_module(). 
move Topology.delete to teardown_module(). 
and split test cases to separate test_functions().